CRM/Contribute/BAO/Query.php

```diff
@@ -149,7 +150,11 @@ public static function where(&$query) {
       if (empty($query->_params[$id][0])) {
         continue;
       }
-      if (substr($query->_params[$id][0], 0, 13) == 'contribution_' || substr($query->_params[$id][0], 0, 10) == 'financial_'  || substr($query->_params[$id][0], 0, 8) == 'payment_') {
+      if (substr($query->_params[$id][0], 0, 13) == 'contribution_'
+        || substr($query->_params[$id][0], 0, 10) == 'financial_'
+        || substr($query->_params[$id][0], 0, 8) == 'payment_'
+        || $query->_params[$id][0] == 'card_type'
```

Comment:
Let's keep the search changes in a separate PR - there are other Prs in the queue that alter all this logic